jira-config/jira-config-platform/src/main/java/org/xwiki/contrib/jira/config/internal/JIRAConfigurationMigrator.java

```diff
@@ -0,0 +1,188 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.jira.config.internal;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.xwiki.bridge.event.ApplicationReadyEvent;
+import org.xwiki.bridge.event.WikiReadyEvent;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.namespace.NamespaceUtils;
+import org.xwiki.extension.job.internal.InstallJob;
+import org.xwiki.job.event.JobFinishedEvent;
+import org.xwiki.model.document.DocumentAuthors;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.LocalDocumentReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.Event;
+import org.xwiki.user.SuperAdminUserReference;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Listener to handle migration of authentication in JIRA configuration.
+ *
+ * @version $Id$
+ * @since 11.0.0
+ */
+@Singleton
+@Component
+@Named(JIRAConfigurationMigrator.HINT)
+public class JIRAConfigurationMigrator extends AbstractEventListener
+{
+    /**
+     * Main name of the listener.
+     */
+    public static final String HINT = "JIRAConfigMigrationListener";
+
+    private static final String JIRA = "JIRA";
+
+    private static final String PROPERTY_USERNAME = "username";
+
+    private static final String PROPERTY_PASSWORD = "password";
+
+    private static final String PROPERTY_ID = "id";
+
+    private static final String PROPERTY_AUTHENTICATION_TYPE = "authenticationType";
+
+    private static final String DOC_SAVE_MESSAGE = "Migrate configration";
+
+    private static final LocalDocumentReference MIGRATION_STATUS_DOCUMENT_REFERENCE =
+        new LocalDocumentReference(JIRA, "JIRAConfigAuthMigration");
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private Logger logger;
+
+    /**
+     * Public constructor.
+     */
+    public JIRAConfigurationMigrator()
+    {
+        super(HINT, List.of(new ApplicationReadyEvent(), new WikiReadyEvent(), new JobFinishedEvent("install")));
```

Comment:
You don't need to go through List.of, there is a `AbstractEventListener(String name, Event... events)`.